pep-0517.txt

```diff
@@ -218,6 +241,14 @@ metadata. The directory passed in by the build frontend MUST be
 identical to the directory created by ``get_wheel_metadata``,
 including any unrecognized files it created.
 
+This method only places the *contents* of the wheel file, without actually
+creating the final artifact. This is to allow flexibility in how the result is
+going to be used. A front end like pip, which is only going to immediately
+install this wheel can speed up the installation process by skipping a round
+trip through zip. However a command like a hypothetical ``twine wheel`` could
```

Comment:
I am strongly -1 on having unpacked wheels in the cache and using links (hard or otherwise). The isolation provided by the current setup is a feature and I don't think adding another flag is worth it in this case.

That being said, we have a few cases that pip is going to be building a wheel, some of which will eventually store the resulting artifact in the wheel cache, but some of which are just going to build an emphereal wheel and then throw it away. Doing something like ``pip install .`` or ``pip install mycoolsdist-1.0.tar.gz`` are not going to store a cached wheel and will rebuild each time.

Beyond that, I could see us adjusting the cache to try and speed it up even more, one example might be to just cache unpacked wheel (but not link them, just essentially precomputing the unzip step), another idea might be to build the wheel files specially to store the files uncompressed inside of the ``.whl`` instead of compresses (obviously not something you'd want to publish to PyPI, but fine for a local cache).

Whether these make sense or not would require benchmarking the size trade off vs the speed won. However building an unpacked wheel basically costs us nothing (except maybe some churn in some projects) over building an already packed wheel. Even if we're in a case that we're ultimately going to zip it up again, we aren't really taking a performance hit or anything, we're just shifting the work from one phase in the process to another, while letting us flexibility to implement performance improvements where they do make sense.